### PR TITLE
compliance: gather compliance requirements for implementing raft

### PR DESCRIPTION
**Pull request by toidiu**
Compliance requirements extracted from the [Raft paper](https://toidiu.com/reads/In_Search_of_an_Understandable_Consensus_Algorithm_(Extended_Raft).pdf).

The goal will be to implement the condensed version of the protocol from Figure 2 in the paper (enumerated in the README.md).

The full set of requirements have been extracted in compliance.md, so it is possible to track if we are missing any important features.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,128 @@
...
-- durable write ahead log
+## Fig 2: condensensed summary of Raft
 
-- role: leader vs follower
-- struct: state_machine { role,  }
-- enum: operations
+### State
+**Persistent state on all servers:**
+- [ ] "Updated on stable storage before responding to RPCs"
+- [ ] currentTerm latest term server has seen (initialized to 0 on first boot,
```
Comment (d-e-s-o) [inline README.md:12]: Nit: Would be nice to use `code highlighting` (with backticks) here for variable names, types etc, to make them stand out more easily.
```suggestion
- [ ] `currentTerm` latest term server has seen (initialized to 0 on first boot,
```

**Review by d-e-s-o — APPROVED** [review]
Thanks for summarizing. Seems useful and I trust it's in line with what the paper outlines, but didn't go over everything again (seems to be mostly a copy of Figure 2). Will refer back to it as I have questions on the code.

File: README.md (hunk 1 above)
Comment (toidiu) [reply README.md:12]: > Also, would be nice to clean up commit history a bit more

Agreed. We should only be doing rebase merge (github ui will let you do that) but you can only enforce that on the paid version of github from what I can tell.

Comment (d-e-s-o) [reply README.md]: > We should only be doing rebase merge (github ui will let you do that) but you can only enforce that on the paid version of github from what I can tell.

Not sure if that's what you mean, but you can go to [Settings](https://github.com/toidiu/raft-rs/settings) and set
- [ ] Allow merge commits Loading
- [ ] Allow squash merging Loading
- [x] Allow rebase merging

**Pull request merged**